text/trace/0107-sampling-priority.md

```diff
@@ -0,0 +1,79 @@
+# Associating sampling priority with the trace using tracestate
+
+Propagating `sampling.priority` (calculated by probability sampler) in a
+`tracestate` will enable a better experience for system with the independent
+stateless configuration of components sampling.
+
+## Motivation
+
+Consistent sampling decision made in each app of a distributed trace is
+important for better user experience of traces analysis. Consistency is achieved
+by aligning stateless hashing algorithm used to make a decision based on
+trace-id or explicitly propagating sampling flag.
+
+Increasingly more apps participate in distributed tracing. With the
+standardized wire format for context propagation, there is bigger chance that
+approach for sampling chosen by one app may not match another app sampling
+policy. Since coordination of sampling policies across many apps not always
+possible, OpenTelemetry must provide a way to share sampling hints between apps.
+
+Beyond sampling hints infrastructure, it is beneficial for the community to
+agree on standard behavior and hints exposed by probability sampler.
+
+The `sampling.priority` property is used by many vendors for various purposes.
+Propagating this field alongside the trace will allow for many improvements and
+as a very minimum will simplify transition of customers from SDKs using
+different `trace-id` based hash functions to OpenTelemetry SDK.
+
+Also see discussion here: <https://github.com/open-telemetry/opentelemetry-specification/pull/570>
+
+## Explanation
+
+### Sampling hints exchange
+
+OpenTelemetry SDK has an infra to expose sampling hints as span attributes today
+via
+[SamplingResult](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample)
+class.
+
+This recommendation suggests OpenTelemetry will allow to return an updated
+`tracestate` that will be used for a newly created `Span`. This will allow
+samplers from a single vendor coordinate sampling decisions across the
+different components of the trace.
+
+### Probability Sampler default behavior
+
+The default behavior of probability sampler is to calculate a stateless hash
+function of `trace-id` to make a sampling decision.
+
+It is suggested to extend the sampler behavior to expose a `tracestate` field
+called `sampling.priority` with the integer value `[0-1000]` that will indicate
```

Comment:
indicate how? What does a value of 408 mean? 40.8% probability?

Reply:
yes. The max value need to be agreed on. 0-1000 would work for most services (sampling down to 1 out of 1000). For extremely loaded services may need more, but likely they may opt out of propagating this flag to optimize performance.

Alternative approach is to go 1..100 and allow non-integers.

Reply:
Large scale systems are _especially_ in need of having one trace stand for many traces; thus I worry about artificially capping at 1000 sample rate.